src/sender.js

```diff
@@ -135,13 +308,10 @@ class Sender {
             if (this.socket) {
                 throw new Error('Sender connected already');
             }
-            this.socket = !secure
-                ? net.connect(options)
-                : tls.connect(options, async () => {
-                    if (!self.socket.authorized) {
-                        reject(new Error('Problem with server\'s certificate'));
-                        await self.close();
-                    }
+            this.socket = !this.secure
+                ? net.connect(connectOptions)
+                : tls.connect(connectOptions, async () => {
+                    await checkServerCert(self, reject);
```

Comment:
Do we have any tests for the failed server cert verification scenario? I'm asking since we're mixing callbacks with async/await here, so it'd be a good idea to make sure that things work as expected.

Reply:
added the test